salt/modules/boto_ec2.py

```diff
@@ -1927,7 +1929,17 @@ def detach_volume(volume_id, instance_id=None, device=None, force=False,
     '''
     conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
     try:
-        return conn.detach_volume(volume_id, instance_id, device, force)
+        ret = conn.detach_volume(volume_id, instance_id, device, force)
+        if wait_to_finish is not None:
+            wait = 0
+            while True:
```

Comment:
Why while Tue here, couldn't you put the condition here directly?

`while wait > wait_to_finish:`